src/en/snowmtl/src/eu/kanade/tachiyomi/extension/en/snowmtl/ComposedImageInterceptor.kt

```diff
@@ -0,0 +1,135 @@
+package eu.kanade.tachiyomi.extension.en.snowmtl
+
+import android.graphics.Bitmap
+import android.graphics.BitmapFactory
+import android.graphics.Canvas
+import android.graphics.Color
+import android.graphics.Paint
+import android.os.Build
+import android.text.Layout
+import android.text.StaticLayout
+import android.text.TextPaint
+import androidx.annotation.RequiresApi
+import kotlinx.serialization.decodeFromString
+import kotlinx.serialization.json.Json
+import okhttp3.Interceptor
+import okhttp3.MediaType.Companion.toMediaType
+import okhttp3.Response
+import okhttp3.ResponseBody.Companion.toResponseBody
+import uy.kohesive.injekt.injectLazy
+import java.io.ByteArrayOutputStream
+import java.io.IOException
+import kotlin.math.min
+
+// The Interceptor joins the captions and pages of the manga.
+@RequiresApi(Build.VERSION_CODES.Q)
+class ComposedImageInterceptor(
+    private val baseUrl: String,
+) : Interceptor {
+
+    private val json: Json by injectLazy()
+
+    override fun intercept(chain: Interceptor.Chain): Response {
+        val request = chain.request()
+        val url = request.url.toString()
+
+        val isPageImageUrl = url.contains("storage.${baseUrl.substringAfterLast("/")}", true)
+        if (isPageImageUrl.not()) {
+            return chain.proceed(request)
+        }
+
+        val translation = request.url.fragment?.parseAs<List<Translation>>()
+            ?: throw IOException("Translation not found")
+
+        val imageRequest = request.newBuilder()
+            .url(url)
+            .build()
+
+        val response = chain.proceed(imageRequest)
+
+        if (response.isSuccessful.not()) {
+            return response
+        }
+
+        val bitmap = BitmapFactory.decodeStream(response.body.byteStream())!!
```

Comment:
I don't expect you do do anything about it, but Suwayomi breaks here. Throwing:

https://github.com/Suwayomi/Suwayomi-Server/blob/fb51834153b33060779e93d3c460f55878a712cb/AndroidCompat/src/main/java/android/graphics/BitmapFactory.java#L21

(It will most likely break on `StaticLayout` too, so probably not even worth looking into.)